src/Products/Traits/With_View_Context.php

```diff
@@ -58,6 +61,115 @@ protected function setup_context( array $alterations = [] ) {
 		return tribe_context()->alter( $alterations );
 	}
 
+	/**
+	 * Builds a "legend" of each event, by post ID.
+	 *
+	 * @return string The event legend.
+	 * @todo move this to the Trait.
+	 *
+	 * The legend should be used to provide more information during debug of failures.
+	 */
+	protected function build_event_legend() {
+		return "Event legend:\n\n" . implode(
+			"\n",
+			array_map(
+				static function ( $event ) {
+					return sprintf(
+						'%d - tz: %s; start: %s; end: %s; all-day: %s',
+						$event->ID,
+						get_post_meta( $event->ID, '_EventTimezone', true ),
+						get_post_meta( $event->ID, '_EventStartDate', true ),
+						get_post_meta( $event->ID, '_EventEndDate', true ),
```

Comment:
Should we have something for UTC as well?

Reply:
I did not want to burden the legend.
Unless we really need to, I would not add it.